SPECIFICATION.md

```diff
@@ -0,0 +1,93 @@
+<h1>Provider Specification</h1>
+
+[TOC]
+
+# General
+- **name** Any record name supplied to the provider must be the full hostname.
+- **ttl** Reasonable default is 6 hours since it's supported by most services. Any service that does not support this must be explicitly mentioned somewhere.
+
+# API Operations
+## create_record
+- **Normal Behavior** Create a new DNS record.
+- **If Record Already Exists** Do nothing. **DO NOT** throw exception.
+- **TTL** If not specified or set to `0`, use reasonable default.
+- **Record Sets** If service has provisions for record sets, **DO NOT** remove any existing records in the set.
+
+## list_record
+- **Normal Behaviour** List all records. Apply filter as required.
+- **Record Sets** Read all record sets and add a record entry as follows:  {name:'', type:'', content: ''}. **DO NOT** bundle together record sets and treat as one record.
+- **Linked Records** For services that support some form of linked record, do not resolve, treat as CNAME.
+
+## update_record
+- **Normal Behaviour** Update a record matching name, type and content.
+- **Record Sets** If matched record is part of a record set, only update the record that matches. **DO NOT** modify other records in any way. Update the record set so that records other than the matched one are unmodified.
+- **TTL**
```

Comment:
I don't think we should provide a way to modify TTL, only content. 
Unfortunately TTL is not supported with all our providers, and if it is, this can be done by deleting a record and re-creating with a new TTL. 

I'm open to feedback here though.

Reply:
I think an update should be able to modify TTL. Then again, deleting and recreating also makes sense. Both are equally valid options.